libraries/rush-lib/src/logic/Telemetry.ts

```diff
@@ -1,13 +1,65 @@
 // Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
 // See LICENSE in the project root for license information.
 
+import * as os from 'os';
 import * as path from 'path';
 import { FileSystem, FileSystemStats, JsonFile } from '@rushstack/node-core-library';
 
 import { RushConfiguration } from '../api/RushConfiguration';
 import { Rush } from '../api/Rush';
 import { RushSession } from '../pluginFramework/RushSession';
 
+/**
+ * @beta
+ */
+export interface ITelemetryMachineInfo {
+  /**
+   * The CPU architecture
+   * @example 'AMD64'
+   */
+  machineArchitecture: string;
+  /**
+   * The CPU
+   */
+  machineCPU: string;
+  /**
+   * The number of logical CPU cores.
+   */
+  machineCores: number;
+  /**
+   * The total amount of RAM on the machine, in MiB.
+   */
+  machineTotalMemoryMiB: number;
+  /**
+   * The amount of free RAM on the machine at the end of execution, in MiB.
+   */
+  machineFreeMemoryMiB: number;
+}
+
+/**
+ * @beta
+ */
+export interface ITelemetryOperationResult {
+  /**
+   * The names of operations that this operation depends on.
+   */
+  dependencies: string[];
+  /**
+   * The status code for the operation.
+   */
+  result: string;
+  /**
+   * A second-resolution timestamp when the operation started.
```

Comment:
So is this a UNIX timestamp, but in seconds? Why not just use a normal UNIX timestamp?

Reply:
I can switch it back to milliseconds, but it's not a UNIX timestamp. It's 0 value is when the process started.